### PR TITLE
Mat 6168 edit definition

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-6168](https://jira.cms.gov/browse/MAT-6168)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitionRow.test.tsx [hunk 1]
```diff
@@ -6,17 +6,24 @@ afterEach(cleanup);
...
+        setOpen={jest.fn()}
+        setSelectedDefinition={jest.fn()}
       />
     );
     const term = getByText("term");
     expect(term).toBeInTheDocument();
     const definition = getByText("I'm a measure definition");
     expect(definition).toBeInTheDocument();
+
+    const editButton = await getByTestId(
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitionRow.test.tsx]: `getByTestId` does not return a promise, so `await` is not needed.

Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitionRow.test.tsx]: @jkotanchik-SB Thanks a lot Joe! Removed.

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx [hunk 2]
```diff
@@ -244,14 +273,18 @@ const MeasureDefinitions = (props: MeasureDefinitionsProps) => {
...
                 </th>
+                <th scope="col" className="col-header">
+                  Action
+                </th>
               </tr>
             </thead>
             <tbody data-testId="measure-definitions-table-body">
               {visibleDefinitions?.length > 0 ? (
                 visibleDefinitions.map((measureDefinition, index) => (
-                  <MeasureMetaDataRow
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx:252]: @mcmcphillips @sb-cecilialiu - To reuse this component, can we not pass the edit function instead?

File: src/components/editMeasure/details/MeasureMetaDataTable.scss [hunk 3]
```diff
@@ -1,65 +0,0 @@
-$border-color: #8c8c8c;
-
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureMetaDataTable.scss]: deleting this file might affect CSS for the reference table?

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx [hunk 4]
```diff
@@ -62,13 +65,21 @@ const MeasureDefinitions = (props: MeasureDefinitionsProps) => {
     measure?.measureSet?.acls,
     measure?.measureMetaData?.draft
   );
+  const sortDefinitions = (measureDefinitions: Array<MeasureDefinition>) => {
+    if (measureDefinitions) {
+      return measureDefinitions.sort((a, b) => a.term.localeCompare(b.term));
+    } else {
+      return [];
+    }
+  };
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx]: `Array.sort` function doesn't create a new array. so you don't need to return anything. I would rather change this function to a comparator for `array.sort`  
e.g.
`const termComparator = (a, b) => a.term.localeCompare(b.term);` 
and wherever needed
`measureDefinitions.sort(termComparator)`

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx (hunk 2 above)
Comment (mcmcphillips) [reply src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx:252]: We could use MeasureMetaDataRow to take in a third variable for function and Add the view logic in a third column

Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx:252]: @adongare @mcmcphillips Changed, thanks!

File: src/components/editMeasure/details/MeasureMetaDataTable.scss (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureMetaDataTable.scss]: @adongare added back.

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx (hunk 4 above)
Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx]: @adongare Changed to use the comparator, thanks!

File: src/components/editMeasure/details/MeasureMetaDataRow.tsx [hunk 5]
```diff
@@ -2,18 +2,52 @@ import React from "react";
...
+    description,
+    measureDefinition,
+    setOpen,
+    setSelectedDefinition,
+  } = props;
+
+  const handleOpen = async (event: React.MouseEvent<HTMLButtonElement>) => {
+    setOpen(true);
+    setSelectedDefinition(measureDefinition);
+  };
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureMetaDataRow.tsx]: I think this should be maintained by the parent component and passed down to this component. we'll have a similar callback for references as well in the future.

**Review by adongare — DISMISSED** [review]

File: src/components/editMeasure/details/MeasureMetaDataRow.tsx (hunk 5 above)
Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureMetaDataRow.tsx]: @adongare setOpen() was also called in the parent component, but setSelectedDefinition() needs to be handled here because the button for edit is in the child component.

Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureMetaDataRow.tsx]: @adongare Thanks so much Ashok!! This is refactored, I just pushed in the new changes.

**Review by adongare — APPROVED** [review]
LGTM

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitionRow.test.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitionRow.test.tsx]: This file was deleted.
